osu.Game/Tests/Visual/OsuTestScene.cs

```diff
@@ -64,7 +68,12 @@ protected IAPIProvider API
 
         protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnlyDependencyContainer parent)
         {
-            Dependencies = new OsuScreenDependencies(false, base.CreateChildDependencies(parent));
+            var baseDependencies = base.CreateChildDependencies(parent);
+
+            if (this is IRulesetTestScene rts)
```

Comment:
I think this is a little bit over-engineeed, and it is probably better that a `CreateRuleset()` method is directly exposed from `OsuTestScene`. Would immediately replace all instances of `CreateRuleset*()` in existing test scenes - there's quite a few already.

@peppy thoughts?

Reply:
and always caching it (with `OsuRuleset` as a sane default)? i'm not necessarily against that.

was unsure about the method used here since it feels like probably too much interface magic for tests

Reply:
I'm not sure if the `OsuRuleset` default is required (thinking forward to splitting all rulesets out into their own repos), but sure.

Reply:
@iiSaLMaN wanna give this a try? removing the interface logic and moving `CreateRuleset` into `OsuTestScene`.

Reply:
Yes sorry, was busy setting up my new MacBook for work yesterday, will get to this.

Reply:
@peppy Although the `OsuTestScene` is a component inside the `osu.Game` assembly, which I think it should never reference any ruleset? will push without the ruleset default.

Reply:
Actually *luckily*, since I've used `Ruleset.Value` as the value of the ruleset, and as it always has the first available ruleset (most likely `OsuRuleset`), the test scene is already technically provides a default ruleset.